Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+        {
+            this.LogMessage("Cleaing out old imports");
+
+            var files = Directory.GetFiles(dir, "DebugRegistryEntries.reg.imported_*.txt").OrderByDescending(f => f);
+            var i = 0;
+            foreach (var file in files)
+            {
+                if (i > 10)
+                {
+                    this.LogMessage($"Deleting {file}");
+                    File.Delete(file);
+                }
+
+                i++;
+            }
+        }
+
+        private void UpdateAddInRegistration()
+        {
+            this.LogMessage("Updating addin registration...");
+            var addInRegFile = Path.Combine(Path.GetDirectoryName(RegFilePath), "RubberduckAddinRegistry.reg");
+            var command = $"reg.exe import \"{addInRegFile}";
+            ExecuteTask(command, ignoreErrors:true);
+        }
+
+        private void ProcessDll(string file)
+        {
+            this.LogMessage($"Processing {file}...");
+
+            var parameters = new DllFileParameters(file, SourceDir, TargetDir);
+            
+            if (_batchPath == string.Empty)
+            {
+                this.LogMessage("Compiling with tlbexp...");
+                CompileWithTlbExp(parameters);
+            }
+            else
+            {
+                this.LogMessage("Compiling with midl...");
+                CreateIdlFile(parameters);
+                CompileWithMidl(parameters);
+            }
+
+            this.LogMessage("Extracting metadata using WiX...");
+            HarvestMetadataWithWix(parameters);
+
+            this.LogMessage("Building registry entries...");
+            var entries = BuildRegistryEntriesFromMetadata(parameters);
+
+            this.LogMessage("Creating InnoSetup registry entries...");
+            CreateInnoSetupRegistryEntries(entries, parameters);
+
+            if (Config != "Debug")
+            {
+                return;
+            }
+
+            RemovePreviousDebugRegistration();
+            UpdateDebugRegistration(entries, parameters);
+        }
+
+        private void CreateIdlFile(DllFileParameters parameters)
+        {
+            var generator = new IdlGenerator();
+            var idl = generator.GenerateIdl(parameters.SourceDll);
+            // Encoding must be UTF8 BOM
+            File.WriteAllText(Path.Combine(TargetDir, parameters.IdlFile), idl, new UTF8Encoding(true));
+        }
+
+        private void CompileWithMidl(DllFileParameters parameters)
+        {
+            var targetPath = TargetDir.TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+            var command = $"call \"{_batchPath}\"{Environment.NewLine}" +
+                          $"midl.exe /win32 /tlb \"{parameters.Tlb32File}\" \"{parameters.IdlFile}\" /out \"{targetPath}\"{Environment.NewLine}" +
+                          $"midl.exe /amd64 /tlb \"{parameters.Tlb32File}\" \"{parameters.IdlFile}\" /out \"{targetPath}\"";
+            ExecuteTask(command, SourceDir, true);
```

Comment:
I liked the named argument usage before. Could we have that here as well for `ignoreErrors`?